_posts/2019-05-03-ortas-best-of.markdown

```diff
@@ -0,0 +1,236 @@
+features.
+
+> Code Review: [Emergence][cr-em], [Energy][cr-en], [Energy Sync][cr-en-sy]
+
+As we started to adopt React Native at Artsy, we really needed a different structure for our entire team and
+technology stack. We had to re-think how we build apps, interact with the open community and what we were as
+engineers.
+
+> [On our implementation of React Native][emis], [Retrospective: Swift at Artsy][retro-swift], [Intro to React
+> Native for an iOS Developer][rn-ios], [React Native, 2 years later][rn2], [Making a React Native Components
+> Pod][rn-pod] & [React Native at Artsy, 3 years later][rn3]
+
+Re-defining ourselves as native engineers who support JavaScript by via our iOS sills was tricky, I think both
+[Ash][ash-on-js] & [Maxim][maxim-talk-culture] have great write-ups on the topic. For me, the move to the
+JavaScript came at a perfect time: the iOS community was fragmenting because of competing dependency managers and
+the introduction of Swift which made infrastructure work less valuable.
+
+We still needed to be up-to-date with the latest tools and ideas from the native world, but mainly from "iOS as
+Platform" instead of features development (though my non-technical [ARKit post][arkit] is a great read)
+
+> [Deploying your app on a weekly basis via fastlane + Travis CI][emis-travis], [What is fastlane match?][match],
+> [It's time to use Swift Package Manager][spm], [Accessing the app's Source Code from your Simulator][src], [Why
+> does my team's Podfile.lock Podspec checksums change?][checksum], [Code Injection for Xcode][inject], [Artsy's
+> first closed source Pod][closed-pod], [CocoaPods-Keys and CI][cpkeys]
+
+# JavaScript
+
+To get experience in JavaScript, I took one of my large open source projects and [re-create it from
+scratch][danger]. Which eventually turned into [Peril][peril], which solves [some interesting
+problems][peril-state] at Artsy and in the rest of my OSS work.
+
+For Artsy, we needed to consider: What is the right tech to build for both React Native and React on web? We had a
+good guess [back in 2016][fejs2017] and that slowly evolved over the course of a few years into what we now call
+the [Artsy Omakase][a-om]. Making sure that the rest of the team agreed, and that new people could go see our
+reasoning was important when making foundations which could last 5-10 years.
+
+> [Exploration: Front-end JavaScript at Artsy in 2017][fejs2017], [Using VS Code for JavaScript][vscode], [GraphQL
+> for iOS Developers][gql-ios], JavaScriptures ([React][j6], [TypeScript][j5], [Tooling][j2], [Relay][j3], [Local
+> State][j2], [Styled Components][j4]), [Why does Artsy use Relay?][why-relay]
+
+# Open Source
+
+Artsy had a rich relationship with Open Source before I arrived, and I devoted a lot of time and effort to making
+this world-class. There is an entire blog post on how Artsy became [Open Source by Default][tttoooodddoooo], but I
```

Comment:
`tttoooodddoooo` should link to https://artsy.github.io/blog/2019/04/29/how-did-artsy-become-oss-by-default/